app/commands.py

```diff
@@ -344,3 +347,24 @@ async def match_start_handler(session: "Session", args: list[str]) -> str | None
             session_id,
             data=match_started_packet,
         )
+
+
+@command("!silence", privileges=ServerPrivileges.CHAT_MODERATOR)
+async def silence_handler(session: "Session", args: list[str]) -> str | None:
+    duration = " ".join(args)
+
+    try:
+        raw_seconds = parse(duration)
+    except:
+        return "Invalid time format! Try again!"
+
+    silence_end = datetime.fromtimestamp(raw_seconds)
```

Comment:
`raw_seconds` is the number of seconds in the timespan you give

so if you say `!silence cmyui 1 day`, `raw_seconds` will be `86400`

Reply:
datetime.fromtimestamp(86400) will give you
```
cmyui@cmyui:~/programming/akatsuki$ python3
Python 3.8.10 (default, Mar 13 2023, 10:26:41)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import datetime
>>> datetime.datetime.fromtimestamp(86400)
datetime.datetime(1970, 1, 1, 19, 0)
```

Reply:
meaning they'll be silenced until `1970, 1, 1`

Reply:
instead, what we want is to get `the current time + timedelta(seconds=86400)`